src/prefect/cli/server.py

```diff
@@ -340,6 +348,20 @@ def start(
     # Pass the Core version so the Server API can return it
     env.update(PREFECT_CORE_VERSION=prefect.__version__)
 
+    if dump:
+        dump_dir = Path(dump)
+        Path(compose_dir_path).joinpath("docker-compose.yml").rename(
+            dump_dir.joinpath("docker-compose.yml")
+        )
+        with dump_dir.joinpath(".env").open("w") as fout:
+            env_vars = [
+                f"{k}={v!r}" if "\n" in v else f"{k}={v}" for k, v in env.items()
```

Comment:
I don't believe docker supports multiline strings at all in its environment file, quoted or not. The doc says it [doesn't treat quotes in any special way](https://docs.docker.com/compose/env-file/), so I doubt using `repr()` on the string would help.

Reply:
Oh that's good to know! I just pulled that code from above presuming you'd `source .env` then run `docker-compose up`. Seems kind of silly that they auto-pull from a `.env` file with their own opinions about syntax but 🤷

Reply:
I'm actually running ` sudo /usr/local/bin/docker-compose --env-file=/opt/python-prefect/docker-compose-env.env up -d` instead of leaving it up to docker to automatically load a hidden file.

And as for the dotenv file syntax, yeah, quoting and multiline is [all over the place](https://github.com/theskumar/python-dotenv/issues/170), that's what tends to happen with loosely-defined formats.

Reply:
Wow that issue is wild. Anyway, I'll make sure to test that it loads from the file correctly :)